src/plugins/vis_type_timeline/server/series_functions/graphite.js

```diff
@@ -35,6 +35,81 @@ import _ from 'lodash';
 import fetch from 'node-fetch';
 import moment from 'moment';
 import Datasource from '../lib/classes/datasource';
+import dns from 'dns-sync';
+import IPCIDR from 'ip-cidr';
+
+const MISS_CHECKLIST_MESSAGE = `Please configure on the opensearch_dashbpards.yml file. 
+You can always enable the default allowlist configuration.`;
+
+const INVALID_URL_MESSAGE = `The Graphite URL/IP provided by you is invalid. 
+Please update your config from OpenSearch Dashboards's Advanced Setting.`;
+
+/**
+ * Resolve hostname to IP address
+ * @param {object} urlObject
+ * @returns {string} configuredIP
+ * or null if it cannot be resolve
+ * According to RFC, all IPv6 IP address needs to be in []
+ * such as [::1]
+ * So if we detect a IPv6 address, we remove brackets
+ */
+function getIpAddress(urlObject) {
+  const hostname = urlObject.hostname;
+  const configuredIP = dns.resolve(hostname);
+  if (configuredIP) {
+    return configuredIP;
+  }
+  if (hostname.startsWith('[') && hostname.endsWith(']')) {
+    return hostname.substr(1).slice(0, -1);
+  }
+  return null;
+}
+
+function isValidURL(configuredUrl, blockedIPs) {
+  // Check the format of URL, URL has be in the format as
+  // scheme://server/path/resource otherwise an TypeError
+  // would be thrown
```

Comment:
let's follow jsdoc convention to describe this function like others.

Reply:
will do.